Error in `kfp-profile-controller` integration tests: Configmaps "kfp-launcher" not found

### Bug Description

This issue has been occurring while working on https://github.com/canonical/kfp-operators/pull/792. The `test_first_change_to_config_for_deafult_pipeline_root` test in the integration tests for `kfp-profile-controller` fails while trying to change the configuration. The relevant error message is:
```
HTTP response body: Error occured while trying to proxy to: localhost:8080/apis/v2beta1/pipelines/upload?name=sample-pipeline
```

#792 doesn't include any changes to that particular test, so the changes introduced shouldn't be the issue here.

A weird aspect of this issue is that it seems to be intermittent, as retriggering the CI without making any changes can make the test pass.


### To Reproduce

.

### Environment

- PR against the main branch
- GitHub runners
- Canonical K8s `1.32`
- Juju `3.6`


### Relevant Log Output

```shell
=================================== FAILURES ===================================
____________ test_first_change_to_config_for_deafult_pipeline_root _____________
Traceback (most recent call last):
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/lightkube/core/generic_client.py", line 235, in raise_for_status
    resp.raise_for_status()
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/httpx/_models.py", line 763, in raise_for_status
    raise HTTPStatusError(message, request=request, response=self)
httpx.HTTPStatusError: Client error '404 Not Found' for url 'https://127.0.0.1:6443/api/v1/namespaces/profilename/configmaps/kfp-launcher'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/_pytest/runner.py", line 344, in from_call
    result: TResult | None = func()
                             ^^^^^^
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/_pytest/runner.py", line 246, in <lambda>
    lambda: runtest_hook(item=item, **kwds), when=when, reraise=reraise
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/pluggy/_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/pluggy/_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py", line 167, in _multicall
    raise exception
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py", line 139, in _multicall
    teardown.throw(exception)
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/_pytest/logging.py", line 850, in pytest_runtest_call
    yield
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py", line 139, in _multicall
    teardown.throw(exception)
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/_pytest/capture.py", line 900, in pytest_runtest_call
    return (yield)
            ^^^^^
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py", line 139, in _multicall
    teardown.throw(exception)
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/_pytest/skipping.py", line 263, in pytest_runtest_call
    return (yield)
            ^^^^^
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/_pytest/runner.py", line 178, in pytest_runtest_call
    item.runtest()
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/_pytest/python.py", line 1671, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/pluggy/_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/pluggy/_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py", line 167, in _multicall
    raise exception
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py", line 139, in _multicall
    teardown.throw(exception)
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py", line 53, in run_old_style_hookwrapper
    return result.get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/pluggy/_result.py", line 103, in get_result
    raise exc.with_traceback(tb)
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py", line 38, in run_old_style_hookwrapper
    res = yield
          ^^^^^
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/pluggy/_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/_pytest/python.py", line 157, in pytest_pyfunc_call
    result = testfunction(**testargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/pytest_asyncio/plugin.py", line 529, in inner
    _loop.run_until_complete(task)
  File "/usr/lib/python3.12/asyncio/base_events.py", line 687, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/home/runner/work/kfp-operators/kfp-operators/charms/kfp-profile-controller/tests/integration/test_charm.py", line 315, in test_first_change_to_config_for_deafult_pipeline_root
    kfp_launcher_configmap = lightkube_client.get(
                             ^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/lightkube/core/client.py", line 230, in get
    return self._client.request("get", res=res, name=name, namespace=namespace)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/lightkube/core/generic_client.py", line 311, in request
    return self.handle_response(method, resp, br)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/lightkube/core/generic_client.py", line 251, in handle_response
    self.raise_for_status(resp)
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.12/site-packages/lightkube/core/generic_client.py", line 237, in raise_for_status
    raise transform_exception(e)
lightkube.core.exceptions.ApiError: configmaps "kfp-launcher" not found
------------------------------ Captured log call -------------------------------
INFO     juju.model:__init__.py:3254 Waiting for model:
  kfp-profile-controller/0 [executing] active: 
INFO     httpx:_client.py:1038 HTTP Request: GET https://127.0.0.1:6443/api/v1/namespaces/profilename/configmaps/kfp-launcher "HTTP/1.1 404 Not Found"
=========================== short test summary info ============================
FAILED tests/integration/test_charm.py::test_first_change_to_config_for_deafult_pipeline_root - lightkube.core.exceptions.ApiError: configmaps "kfp-launcher" not found
=================== 1 failed, 10 passed in 367.87s (0:06:07) ===================
integration: exit 1 (371.62 seconds) /home/runner/work/kfp-operators/kfp-operators/charms/kfp-profile-controller> pytest -vv --tb native --asyncio-mode=auto /home/runner/work/kfp-operators/kfp-operators/charms/kfp-profile-controller/tests/integration --log-cli-level=INFO -s --model kubeflow --charm-path=/home/runner/work/kfp-operators/kfp-operators/charms/kfp-profile-controller/kfp-profile-controller_ubuntu@24.04-amd64.charm pid=101486
  integration: FAIL code 1 (382.42=setup[5.83]+cmd[4.97,371.62] seconds)
  evaluation failed :( (382.44 seconds)
kfp-profile-controller-integration: exit 1 (382.67 seconds) /home/runner/work/kfp-operators/kfp-operators> tox -c charms/kfp-profile-controller -e integration -- --model kubeflow --charm-path=/home/runner/work/kfp-operators/kfp-operators/charms/kfp-profile-controller/kfp-profile-controller_ubuntu@24.04-amd64.charm pid=101052
  kfp-profile-controller-integration: FAIL code 1 (390.51=setup[7.84]+cmd[382.67] seconds)
  evaluation failed :( (390.56 seconds)
```

### Additional Context

Perhaps the issue could be because of the changes introduced in #787

--- a/charms/kfp-profile-controller/tests/integration/test_charm.py
+++ b/charms/kfp-profile-controller/tests/integration/test_charm.py
@@ -66,7 +66,7 @@ def wait_for_configmap(client: lightkube.Client, name: str, namespace: str) -> C
     for attempt in RETRY_FOR_ONE_MINUTE:
         with attempt:
             return client.get(res=ConfigMap, name=name, namespace=namespace)
-    raise TimeoutError(f"ConfigMap {name} not present.")
+    raise TimeoutError(f"ConfigMap {name} in namespace {namespace} not present.")
 
 
 @pytest.mark.abort_on_fail

